src/Models/HydrostaticFreeSurfaceModels/HydrostaticFreeSurfaceModels.jl

```diff
@@ -75,9 +75,19 @@ Return a flattened `NamedTuple` of the fields in `model.velocities`, `model.free
 `model.tracers`, and any auxiliary fields for a `HydrostaticFreeSurfaceModel` model.
 """
 @inline fields(model::HydrostaticFreeSurfaceModel) = 
-    merge(hydrostatic_fields(model.velocities, model.free_surface, model.tracers),
-          model.auxiliary_fields,
-          biogeochemical_auxiliary_fields(model.biogeochemistry))
```

Comment:
I am a bit confused about this function. I think that the biogeochemical auxiliary fields have been added to the `auxiliary_fields` during model construction, so we shouldn't need an extra `biogeochemical_auxiliary_fields(model.biogeochemistry)` here right? @jagoosw is it true?

Reply:
no, the biogeochemistry struct is allowed to have extra auxiliary fields that are not attached to the model

Reply:
It actually looks like the fields are added to the `auxiliary_fields`, and, thus they are accounted for twice:
https://github.com/CliMA/Oceananigans.jl/blob/d8273e7a07e8f53f9772f87f67f6eca8d17cb98c/src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_model.jl#L131-L133